### ISSUE TITLE
EIP-4844 Implementers' Call #1

### ISSUE THREAD

**Issue by timbeiko** [ethereum/pm#647]
## Meeting Info
- [Oct 25, 2022, 14:30 UTC](https://savvytime.com/converter/utc-to-germany-berlin-united-kingdom-london-ny-new-york-city-ca-san-francisco-china-shanghai-japan-tokyo-australia-sydney/oct-25-2022/14-30)
- Duration: 60 minutes
- Recording: https://youtu.be/hAa1b4m7tsI 
- [Notes & action items](https://docs.google.com/document/d/15EatedrJanNxBZGPVASvwq9xgbTs5UxjsDfjpM6ppSY/edit?usp=sharing) 
- Zoom: posted in `#4844-testing` prior to the call 

📆  Subscribe to the [Ethereum Protocol Call calendar](https://calendar.google.com/calendar/u/1?cid=Y191cGFvZm9uZzhtZ3JtcmtlZ243aWM3aGs1c0Bncm91cC5jYWxlbmRhci5nb29nbGUuY29t) for calendar invites

## Agenda
- Spec updates 
    - https://github.com/ethereum/EIPs/pull/5707 
    - [Exposing the modulus to the EVM](https://github.com/ethereum/pm/issues/647#issuecomment-1284584357)
    - https://github.com/ethereum/consensus-specs/pull/3046 
    - https://github.com/ethereum/consensus-specs/pull/3047
    - https://github.com/ethereum/consensus-specs/pull/3038 
- Implementation / devnet updates 
    - https://github.com/ethereum/consensus-specs/pull/3052
    - kzg libraries for all client teams  
    - Client status updates 
- Testing / tooling updates
    - 2MB Block Testings  
- [Readiness checklist](https://github.com/ethereum/pm/blob/master/Breakout-Room/4844-readiness-checklist.md)
- Other

**Comment by vbuterin** [comment]
We should expose the modulus to the EVM in some way. This is necessary if we want rollups that work with the point evaluation precompile to be forward-compatible with future changes that change the EVM. In particular, I expect a significant chance that we'll change the commitment scheme over to STARKed Merkle roots over a 64-bit modulus (!!) as hashes and STARKs are very fast in that context. If the modulus is available in the EVM, rollups would be able to write their proof-of-equivalence circuits to work with whatever modulus is provided, and would be forward-compatible.

A few possible options include:

* A `MODULUS` opcode that returns the current modulus
* A `MODULUS` opcode that returns the modulus associated with an inputted version byte
* Modifying the point evaluation precompile to also take as input the modulus as an argument, and fail if the provided modulus does not match that used in the provided commitment
* Modifying the point evaluation precompile to output the modulus used in the given commitment, if evaluation is successful

The last option probably requires the fewest client changes. But whatever we want to do regarding this, it's probably worth agreeing on the exact approach and adding it to the EIP soon.

**Comment by terencechain** [comment]
I won't be able to make this due to a conflict, but there are prysm updates:
- Cleaning up https://github.com/terencechain/prysm/pull/1 into an excellent state to be merged into our canonical master branch
- Raul is looking into storing blobs in the DB with keys as a circular buffer with a length equal to expiration. The current two buckets approach won't work well because bolt freelist is troublesome. We've had a bad experience with slasher db

Spec updates, I opened the following. Thanks everyone for the feedback so far!
https://github.com/ethereum/consensus-specs/pull/3046
https://github.com/ethereum/consensus-specs/pull/3047

2mb testing, I opened this branch to be able to capture attestation arrival latency:
https://github.com/prysmaticlabs/prysm/compare/2mb?expand=1
I'm looking for clarification on whether we need to capture individual attestation latency in DB + additional API support
